### ISSUE TITLE
Derivatives of CSV input

### ISSUE THREAD

**Issue by AnHeuermann** [OpenModelica/OpenModelica#9837]
### Description

Derivatives of top-level CSV input are currently not supported by OpenModelica. With #9804 at least an error is thrown and the compiler stops with a somewhat useful message.

But the runtime should be able to compute the derivative from the linear interpolation used for CSV inputs.

### Steps to Reproduce

Create an input CSV file representing `time^2`
```csv
"time","Modelica.Blocks.Continuous.Der.u"
0,0
0.01,0.0001
0.02,0.0004
0.03,0.0009
0.04,0.0016
0.05,0.0025
[...]
```
[input.csv](https://github.com/OpenModelica/OpenModelica/files/10140610/input.csv)

And simulate model
```modelica
block Modelica.Blocks.Continuous.Der "Derivative of input (= analytic differentiations)"
  input Real u "Connector of Real input signal";
  output Real y "Connector of Real output signal";
equation
  y = der(u);
end Modelica.Blocks.Continuous.Der;
```
with `-csvInput=input.csv`.

### Expected Behavior

Compute derivative numerically from interpolated values. `startTime` and `stopTime` need to be handled as well.

### Version and OS

 - OpenModelica Version: v1.21.0-dev-99-g6a23e6cf85-cmake
 - OS: Windows 11

### Additional Context

From https://github.com/OpenModelica/OpenModelica/issues/9802.

**Comment by casella** [comment]
@AnHeuermann I'm not 100% sure this is safe. What if the CSV input contains discontinuous behaviour, e.g.

```
"time","Modelica.Blocks.Continuous.u"
0,0
0.01,0.0001
0.02,0.0004
0.02,0.0006
0.03,0.0009
0.04,0.0016
0.05,0.0025
```

**Comment by phannebohm** [comment]
Piecewise linear functions are not differentiable at the sampling points, so `der(u)` would be piecewise constant. Should we then trigger a time event at each sampling point?

BTW `u` is considered known and I guess `der(u)` would also be known, so the CSV input containing discontinuous behaviour would not be a problem per se, it would just trigger a time event.

**Comment by casella** [comment]
@phannebohm I'm afraid you're missing the point.

If the piecewise linear input is _continuous_, then it is also differentiable, except in the corner point, where the derivative has a different left and right limit but is not defined at the point. Generating events at each sampling point would guarantee a more numerically accurate result if you are using a high-order integration algorithm, but it will slow down the simulation. Whether or not this is a good idea, I'm not sure. The story is exactly the same as the handling of `smooth(0, <if-expression>)` expressions, should we trigger events when the condition of the if-expression changes? I think we currently don't do that, and I'm fine with that, but this is probably questionable if you use a 4-th order integration algorithm.

If the piecewise linear input is _discontinuous_, then the derivative at that point is not simply discontinuous - it is a Dirac delta-function, which we cannot represent in our context. So, the model is invalid. Completely ignoring impulses doesn't seem a good idea to me.

For example, consider this tank model, where you force the level to be an input, hence obtaining an index-2 problem

```modelica
model Bad
  input Real level;
  Real flowRate;
  parameter Real Area;
equation
  Area*der(level) = flowRate;
end Bad;
```
If the piecewise linear level is continuous, then you can figure out the flow rate, which is just the piecewise constant derivative of the input. No problem. However, if the piecewise linear level is discontinuous at time = t0, there is no way you can compute a (finite) flow rate that gets you this instantaneous change of level - the flow rate should have an impulse at time = t0. Ignoring that would give a completely misleading result.

To me, the safer solution is what we have now, i.e., abort the translation if top-level inputs need to be differentiated. If we don't, we should at least check at run time that we never have two input values at the same time. If we do, the simulation should abort at run time with error: "Cannot differentiate discontinuous input".

**Comment by phannebohm** [comment]
Makes sense. And this is probably in accord with other scenarios involving derivatives of potentially discontinuous signals. I think we have some open issues on that.

EDIT: here it is #8658

**Comment by casella** [comment]
We should also consider this for FMU generatio. In this case I would play it safe and always give an error if inputs must be differentiated. That really makes no sense for FMUs IMHO.
